esp/bindings/http/platform/httpbinding.cpp

```diff
@@ -1277,14 +1277,28 @@ static void filterXmlBySchema(IPTree* in, IXmlType* type, const char* tag, Strin
 
     if (type->isComplexType())
     {
+        const char *typeName = type->queryName();
         setStartTag(in,type,tag,out);
 
         int flds = type->getFieldCount();
         for (int i=0; i<flds; i++)
         {
             const char* fldName = type->queryFieldName(i);
             IPTree* fld = in ? in->queryBranch(fldName) : NULL;
-            filterXmlBySchema(fld,type->queryFieldType(i),fldName,out);
+            IXmlType* fldType = type->queryFieldType(i);
+
+            //normally even if the tag wasn't in the xml we would continue to put empty tags or defaults in place
+            //  but if the field is a recursive reference we wan't to stop here
```

Comment:
very minor spelling issue: "wan't"
but while we're here, I was wondering if it's actually necessary to test for !fld in Line 1292

Reply:
fld is a pointer to the data that was entered.

This is saying, "continue processing recursively as long as there is child data (fld), and then stop if there is not (!fld)".

If I didn't check for !fld I think it would stop at the first instance of the structure and not process any child data for those fields at all.